src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs

```diff
@@ -191,21 +191,24 @@ private RequestDelegate BuildApplication()
 
                 return builder.Build();
             }
-            catch (Exception ex) when (_options.CaptureStartupErrors)
+            catch (Exception ex)
             {
                 // EnsureApplicationServices may have failed due to a missing or throwing Startup class.
                 if (_applicationServices == null)
                 {
                     _applicationServices = _applicationServiceCollection.BuildServiceProvider();
                 }
 
-                EnsureServer();
-
                 // Write errors to standard out so they can be retrieved when not in development mode.
                 Console.Out.WriteLine("Application startup exception: " + ex.ToString());
                 var logger = _applicationServices.GetRequiredService<ILogger<WebHost>>();
                 logger.ApplicationError(ex);
 
+                if (!_options.CaptureStartupErrors)
+                    throw;
+
+                EnsureServer();
```

Comment:
It wasn't called before when CaptureStartupErrors=false. Since it re-throws, there's no need to create the server.